### PR TITLE
{lang}[GCCcore/11.3.0] RStudio-Server v2022.07.1+554

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

This would benefit from https://github.com/easybuilders/easybuild-framework/pull/4073

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 5 out of 6 (5 easyconfigs in total)
taurusa11 - Linux CentOS Linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), 3 x NVIDIA GeForce GTX 1080 Ti, 460.32.03, Python 2.7.5
See https://gist.github.com/e5bf145d6a010d824ededba6e1d33e3d for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
taurusa14 - Linux CentOS Linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), 3 x NVIDIA GeForce GTX 1080 Ti, 460.32.03, Python 2.7.5
See https://gist.github.com/d46dc1c0df7aff0fc64fb6fa2f595629 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
taurusa8 - Linux CentOS Linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), 3 x NVIDIA GeForce GTX 1080 Ti, 460.32.03, Python 2.7.5
See https://gist.github.com/dbe5b65cc9d7b1ca783ae96b588fd9b6 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
taurusa8 - Linux CentOS Linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), 3 x NVIDIA GeForce GTX 1080 Ti, 460.32.03, Python 2.7.5
See https://gist.github.com/a290ec6e8d8b2efa65dbd9c8ec77cae8 for a full test report.

**Review by jfgrimm — COMMENTED** [review]
As decided in issue #16330, we have deprecated the use of `True` to signify a system-toolchain dependency (#16384), in favour of the more intuitive `SYSTEM` template constant. Due to the change in the test suite, please run `eb --sync-pr-with-develop 15967` and update the PR to use `SYSTEM` instead.

**Comment by Flamefire** [comment]
Rebased and integrated the change. I know about that but this has been open for so long now, it hasn't been decided yet when the PR was opened

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 3 out of 4 (3 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/f3a938548057381f347c424933d73970 for a full test report.

**Comment by Flamefire** [comment]
>     Sanity check failed: sanity check command MYTMP=`mktemp -d` && echo -e "provider=sqlite\ndirectory=$MYTMP/db" >> "$MYTMP/db.conf" && rserver --verify-installation=1 --server-user="$USER" --database-config-file="$MYTMP/db.conf" --server-data-dir="$MYTMP/sdd" exited with code 1 (output: )

Not output but exit code 1. Any idea? Can you try to run it manually and see what fails?

**Comment by verdurin** [comment]
Here's what I see from trying to run that sanity check manually:
```
MYTMP=`mktemp -d` && echo -e "provider=sqlite\\ndirectory=$MYTMP/db" >> "$MYTMP/db.conf" && ./rserver --verify-installation=1 --server-user="$USER" --database-config-file="$MYTMP/db.conf" --server-data-dir="$MYTMP/sdd"
TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/etc/rstudio/logging.conf'. Logging to '/home/software/.local/share/rstudio/log/rserver.log'.
2022-10-18T12:17:29.973884Z [rserver] ERROR system error 2 (No such file or directory) [path: /tmp/rstudio-server/secure-cookie-key]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForWrite(std::shared_ptr<std::basic_ostream<char> >&, bool) const src/cpp/shared_core/FilePath.cpp:1520; LOGGED FROM: rstudio::core::Error rstudio::core::{anonymous}::openFileForWritingWithRetry(const rstudio::core::FilePath&, bool, int, std::shared_ptr<std::basic_ostream<char> >*) src/cpp/core/FileSerializer.cpp:73
2022-10-18T12:17:29.974257Z [rserver] ERROR system error 2 (No such file or directory) [path: /tmp/rstudio-server/secure-cookie-key]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForWrite(std::shared_ptr<std::basic_ostream<char> >&, bool) const src/cpp/shared_core/FilePath.cpp:1520; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:713
```

**Comment by Flamefire** [comment]
@verdurin There is a duplicated slash in your command. It should be    
`MYTMP=`mktemp -d` && echo -e "provider=sqlite\ndirectory=$MYTMP/db" >> "$MYTMP/db.conf" && rserver --verify-installation=1 --server-user="$USER" --database-config-file="$MYTMP/db.conf" --server-data-dir="$MYTMP/sdd"`

Check the created `db.conf` file that it looks like this:
```
provider=sqlite
directory=/tmp/foobar/db
```

**Comment by verdurin** [comment]
Here you go:
```
MYTMP=`mktemp -d` && echo -e "provider=sqlite\ndirectory=$MYTMP/db" >> "$MYTMP/db.conf" && ./rserver --verify-installation=1 --server-user="$USER" --database-config-file="$MYTMP/db.conf" --server-data-dir="$MYTMP/sdd"
TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/etc/rstudio/logging.conf'. Logging to '/home/software/.local/share/rstudio/log/rserver.log'.
2022-10-18T12:36:36.319932Z [rserver] ERROR system error 2 (No such file or directory) [path: /tmp/rstudio-server/secure-cookie-key]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForWrite(std::shared_ptr<std::basic_ostream<char> >&, bool) const src/cpp/shared_core/FilePath.cpp:1520; LOGGED FROM: rstudio::core::Error rstudio::core::{anonymous}::openFileForWritingWithRetry(const rstudio::core::FilePath&, bool, int, std::shared_ptr<std::basic_ostream<char> >*) src/cpp/core/FileSerializer.cpp:73
2022-10-18T12:36:36.320690Z [rserver] ERROR system error 2 (No such file or directory) [path: /tmp/rstudio-server/secure-cookie-key]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForWrite(std::shared_ptr<std::basic_ostream<char> >&, bool) const src/cpp/shared_core/FilePath.cpp:1520; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:713
[software@easybuild-c7 bin]$ cat $MYTMP/db.conf
provider=sqlite
directory=/tmp/tmp.7C6kfikcwe/db
```

**Comment by Flamefire** [comment]
Thanks, that helped finding the issue. Try with 

    MYTMP=`mktemp -d` && echo -e "provider=sqlite\ndirectory=$MYTMP/db" >> "$MYTMP/db.conf" && ./rserver --verify-installation=1 --server-user="$USER" --database-config-file="$MYTMP/db.conf" --server-data-dir="$MYTMP/sdd" --secure-cookie-key-file="$MYTMP/secure-cookie-key"

And if that works, run the sanity check (only) of the updated PR (or a full rebuild if you like)

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 49 out of 51 (3 easyconfigs in total)
centos7.lan - Linux CentOS Linux 7.9.2009, x86_64, Intel Core Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/086e6c2d0819b1a3244671f94ce4e17c for a full test report.

**Comment by verdurin** [comment]
A different error this time:
```
 MYTMP=`mktemp -d` && echo -e "provider=sqlite\ndirectory=$MYTMP/db" >> "$MYTMP/db.conf" && ./rserver --verify-installation=1 --server-user="$USER" --database-config-file="$MYTMP/db.conf" --server-data-dir="$MYTMP/sdd" --secure-cookie-key-file="$MYTMP/secure-cookie-key"

TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/etc/rstudio/logging.conf'. Logging to '/home/software/.local/share/rstudio/log/rserver.log'.

2022-10-19T08:51:12.834588Z [rserver] ERROR system error 2 (No such file or directory) [path: /tmp/rstudio-server/session-rpc-key]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForWrite(std::shared_ptr<std::basic_ostream<char> >&, bool) const src/cpp/shared_core/FilePath.cpp:1520; LOGGED FROM: rstudio::core::Error rstudio::core::{anonymous}::openFileForWritingWithRetry(const rstudio::core::FilePath&, bool, int, std::shared_ptr<std::basic_ostream<char> >*) src/cpp/core/FileSerializer.cpp:73

2022-10-19T08:51:12.834940Z [rserver] ERROR system error 2 (No such file or directory) [path: /tmp/rstudio-server/session-rpc-key]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForWrite(std::shared_ptr<std::basic_ostream<char> >&, bool) const src/cpp/shared_core/FilePath.cpp:1520; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:786
```

**Comment by Flamefire** [comment]
Ok worked on this again and tried with a readonly `/tmp/rstudio-server` directory. Should now be fixed.
If that requires even more to be done for the check then we likely need an easyblock ;-)

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (3 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/a340934bb0f1136e7c84fad2b72d9174 for a full test report.

**Comment by verdurin** [comment]
@Flamefire thanks for your efforts here. Could you send an updated build from your site too?

**Comment by kosl** [comment]
I suggest to upgrade this PR with #16426 and take useful module help. 
Otherwise, excellent EB and patches.

**Comment by Flamefire** [comment]
> I suggest to upgrade this PR with https://github.com/easybuilders/easybuild-easyconfigs/pull/16426 and take useful module help.

@kosl Sorry I don't understand, what should I take from there?
Or do you just mean that this PR should use the newer version v2022.07.2-576?

@verdurin This PR has been open for a while. I could submit a new test, a bit reluctant upgrading the version as I extensively tested this with the help of others in Slack. Anyway: Which version do we want to go with and does anything else needs changing adding?

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
taurusa8 - Linux CentOS Linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), 3 x NVIDIA GeForce GTX 1080 Ti, 460.32.03, Python 2.7.5
See https://gist.github.com/82e4c2f5ea51a71d08786adf5cf28637 for a full test report.

File: easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2022.07.1+554-foss-2022a-Java-11-R-4.2.1.eb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+version = '2022.07.1+554'
+versionsuffix = '-Java-%(javaver)s-R-%(rver)s'
+local_git_rev = '7872775ebddc40635780ca1ed238934c3345c5de'
+
+homepage = 'https://www.rstudio.com/'
+description = """This is the RStudio Server version.
+RStudio is a set of integrated tools designed to help you be more productive with R.
+
+The server can be started with:
+  rserver --server-daemonize=0 --www-port 8787 --rsession-which-r=$(which R)
```
Comment (verdurin) [inline easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2022.07.1+554-foss-2022a-Java-11-R-4.2.1.eb:13]: I agree with @kosl that the longer invocation text is better.

```suggestion
  rserver --server-daemonize=0 --www-port=8787 --rsession-which-r=$(which R) 
--server-user=${USER} --secure-cookie-key-file=${MYTMP}/secure-cookie-key 
--server-data-dir=${MYTMP}/sdd --database-config-file=${MYTMP}/db.conf
```

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2022.07.1+554-foss-2022a-Java-11-R-4.2.1.eb (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2022.07.1+554-foss-2022a-Java-11-R-4.2.1.eb:13]: Why? IMO that is to much noise with many parameters which are not required. For example the default paths for user, coockie-key-file etc are fine, some already in a /tmp and maybe even more useful than in a newly created dir which could cause issues when accidentally starting multiple servers. I'd expect users to know better how to configure the server and if not there is documentation for this.

**Comment by Flamefire** [comment]
Also PR for the newer version: #16478

File: easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2022.07.1+554-foss-2022a-Java-11-R-4.2.1.eb (hunk 1 above)
Comment (kosl) [reply easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2022.07.1+554-foss-2022a-Java-11-R-4.2.1.eb]: User that would like to start rserver on a compute node needs all above parameters. The package is not meant for sysadmin only.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/db3537e51edbc74e13d00731047f3216 for a full test report.

**Comment by Flamefire** [comment]
As #16478 is now merged, I guess this can be closed? Or is there interest in having this?

**Pull request closed without merging**